core/Pos/Binary/Core/Delegation.hs

```diff
@@ -1,16 +1,32 @@
 -- | Delegation types serialization.
 
-module Pos.Binary.Core.Delegation
-       (
-       ) where
+module Pos.Binary.Core.Delegation () where
 
+import           Nub (ordNub)
 import           Universum
 
+import qualified Data.Set as S
+
 import           Pos.Binary.Class (Bi (..))
+import           Pos.Binary.Core.Slotting ()
 import           Pos.Binary.Crypto ()
-import           Pos.Core.Configuration (HasConfiguration)
-import           Pos.Core.Delegation (DlgPayload (..))
+import           Pos.Core.Delegation (DlgPayload (..), HeavyDlgIndex (..), LightDlgIndices (..),
+                                      ProxySKHeavy)
+
+instance Bi HeavyDlgIndex where
+    encode = encode . getHeavyDlgIndex
+    decode = HeavyDlgIndex <$> decode
+
+instance Bi LightDlgIndices where
+    encode = encode . getLightDlgIndices
+    decode = LightDlgIndices <$> decode
 
-instance HasConfiguration => Bi DlgPayload where
-    encode = encode . getDlgPayload
-    decode = UnsafeDlgPayload <$> decode
+instance Bi DlgPayload where
+    encode = encode . S.toList . getDlgPayload
+    decode = do
+        (psks :: [ProxySKHeavy]) <- decode
+        let asSet :: Set ProxySKHeavy
+            asSet = S.fromList psks
+        when (length psks /= length (ordNub psks)) $
```

Comment:
`length` is `O(1)` for `Set`, so I would use `length asSet` instead of `length (ordNub psks)`. In most cases `asSet` will be forced, so it should be more efficient (and clear).

Reply:
Fixed.